tempesta_fw/http.c

```diff
@@ -2738,21 +2737,25 @@ tfw_http_req_process(TfwConn *conn, const TfwFsmData *data)
 			       && (req->content_length != req->body.len));
 		}
 
-		/* Assign the right Vhost for this request. */
-		if (tfw_http_req_set_context(req)) {
-			TFW_INC_STAT_BH(clnt.msgs_otherr);
-			tfw_client_drop(req, 500, "cannot find"
-				      "Vhost for request");
-			return TFW_BLOCK;
+		/* Assign the right Vhost for current request. */
+		if ((req->vhost = tfw_http_tbl_vhost((TfwMsg *)req, &block))) {
+			req->location = tfw_location_match(req->vhost,
+							   &req->uri_path);
+			r = tfw_gfsm_move(&conn->state, TFW_HTTP_FSM_REQ_MSG,
```

Comment:
The hook is not called if vhost is not found. That is not right. Need to run the hook even if vhost is not found, default vhost and default  location will be used as context.

Currently only frang is behind that hook, it still may want to block the client even if no vhost found.

Reply:
Yeah, good catch - this must be fixed before the merge.

Reply:
Corrected.